libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -2127,6 +2130,65 @@ void GCS_MAVLINK::send_raw_imu()
 #endif
 }
 
+#if AP_MAVLINK_MSG_HIGHRES_IMU_ENABLED
+void GCS_MAVLINK::send_highres_imu()
+{
+    uint16_t fields_updated = 0;
+    Vector3f accel;
+    Vector3f gyro;
+#if AP_INERTIALSENSOR_ENABLED
+    const AP_InertialSensor &ins = AP::ins();
+    accel = ins.get_accel();
+    gyro = ins.get_gyro();
+    fields_updated |= 0x3F;
+#endif
+    
+    Vector3f mag;
+#if AP_COMPASS_ENABLED
+    const Compass &compass = AP::compass();
+    if (compass.get_count() >= 1) {
+        mag = compass.get_field() * 1000.0f; // convert to gauss
+    }
+    fields_updated |= 0xC0;
+#endif
+
+    float press_abs = 0.0f;
+    float temperature = 0.0f;
+    float altitude = 0.0f;
+    float diff_pressure = 0.0f;
+#if AP_BARO_ENABLED
+    const AP_Baro &barometer = AP::baro();
+    press_abs = barometer.get_pressure() * 0.01f;
+    temperature = barometer.get_temperature();
+    altitude = barometer.get_altitude();
+    diff_pressure = press_abs - barometer.get_ground_pressure() * 0.01f;
+    fields_updated |= 0xF00;
+#endif
+    // set to HIGHRES_IMU_UPDATED_ALL 
+    fields_updated = fields_updated == 0x0FFFF ? 0xFFFF : fields_updated;
+
+    mavlink_msg_highres_imu_send(
+        chan,
+        AP_HAL::micros64(),
+        accel.x,
+        accel.y,
+        accel.z,
+        gyro.x,
+        gyro.y,
+        gyro.z,
+        mag.x,
+        mag.y,
+        mag.z,
+        press_abs,
+        diff_pressure,
+        altitude,
+        temperature,
+        fields_updated,
+        0
```

Comment:
why not loop through all IMUs instead of only sending the first instance?

Reply:
Okay, I would implement it. But how would you interpret this with regards to the barometer and magnetometer? There could be multiple instaces, too.
How would you suggest going about this? Looping IMUs, Baros and compasses separately?

Reply:
I'd say just loop through all the indexes. We usually have a max of 3 of each so just do that
IMU1, Mag1, Baro1, Temp 1
IMU2, Mag2,  Baro2, Temp2
IMU3, Mag3,  Baro3, Temp3

If there is no mag or baro or whatever, the data should be 0 and the fields_updated bits not set